### PR TITLE
secure by blocking access to page if the user is not logged

### LINKED ISSUE #141: As a user, I want to have access only to what I am entitled to, in order to have a safe navigation.

### LINKED ISSUE #142: secure access to the pages so that a user who is not logged in cannot go there and is redirected to the homepage

### PR DESCRIPTION
**Pull request by jlstoupy98**
some changes to secure a bit the app & improve coverage

### CODE REVIEW

File: app/src/androidTest/java/com/sdp/movemeet/Activity/ActivityNotNullDescriptionTest.java [hunk 1]
```diff
@@ -54,13 +69,51 @@
...
+            }
+        });
+
+        try {
+            latch.await();
+        } catch (InterruptedException e) {
+            assert (false);
+        }
+
+        ActivityScenarioRule<ActivityDescriptionActivity> testRule = new ActivityScenarioRule<>(new Intent(getApplicationContext(), ActivityDescriptionActivity.class).putExtra("activity", activity));
```
Comment (Antho1426) [inline app/src/androidTest/java/com/sdp/movemeet/Activity/ActivityNotNullDescriptionTest.java:95]: So you can navigate to ActivityDescriptionActivity in the middle of a test like this? This interests me because I'm trying to do something similar with my test for deleting an account:
1. Create an account in RegisterActivity
2. Jump to ProfileActvitiy and click the "Delete Account" button

File: app/src/androidTest/java/com/sdp/movemeet/Activity/UploadActivityActivityTest.java [hunk 2]
```diff
@@ -113,7 +110,7 @@ public void onComplete(@NonNull Task<AuthResult> task) {
     @Test
     @LargeTest
     public void endToEnd() {
-        ActivityScenario scenario = testRule.getScenario();
+        ActivityScenario scenario = ActivityScenario.launch(UploadActivityActivity.class);
```
Comment (Antho1426) [inline app/src/androidTest/java/com/sdp/movemeet/Activity/UploadActivityActivityTest.java:113]: Why did you change this line? What is the benefit of "ActivityScenario.launch(UploadActivityActivity.class)" over "testRule.getScenario()" according to you?

File: app/src/androidTest/java/com/sdp/movemeet/Login/LoginActivityTest.java [hunk 3]
```diff
@@ -77,41 +77,24 @@
...
+    }
+
     @After
     public void signOut() {
         FirebaseAuth fAuth = FirebaseAuth.getInstance();
         if (fAuth.getCurrentUser() != null) {
             fAuth.signOut();
         }
     }
-
```
Comment (Antho1426) [inline app/src/androidTest/java/com/sdp/movemeet/Login/LoginActivityTest.java:91]: Removing some tests, okay ahah 😝

File: app/src/androidTest/java/com/sdp/movemeet/Navigation/ActivityDescriptionNavigationTest.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        onView(withId(R.id.drawer_layout))
+                .check(matches(isClosed(Gravity.LEFT))) // Left Drawer should be closed.
+                .perform(DrawerActions.open()); // Open Drawer
+
+        try{
+            Thread.sleep(500);
+        }catch(Exception e){}
+
+        onView(withId(R.id.nav_view)).perform(NavigationViewActions.navigateTo(R.id.nav_logout));
+    }
```
Comment (Antho1426) [inline app/src/androidTest/java/com/sdp/movemeet/Navigation/ActivityDescriptionNavigationTest.java:112]: Okay, this is what was previously MainActivityTest, right?

File: app/src/androidTest/java/com/sdp/movemeet/Navigation/ChatActivityNavigationTest.java [hunk 5]
```diff
@@ -0,0 +1,113 @@
+package com.sdp.movemeet.Navigation;
+
+import android.view.Gravity;
```
Comment (Antho1426) [inline app/src/androidTest/java/com/sdp/movemeet/Navigation/ChatActivityNavigationTest.java:3]: Okay, I see, you had to create such test for each page accessible from the side panel to navigate to each of the other pages!

File: app/src/androidTest/java/com/sdp/movemeet/Security/ProfileActivitySecurityTest.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+@RunWith(AndroidJUnit4.class)
+
+public class ProfileActivitySecurityTest {
+
+    @Rule
+    public ActivityScenarioRule<ProfileActivity> testRule = new ActivityScenarioRule<>(ProfileActivity.class);
+
+    @Test
+    public void redirectionTest() {
+        onView(withId(R.id.button_login)).check(matches(isDisplayed()));
```
Comment (Antho1426) [inline app/src/androidTest/java/com/sdp/movemeet/Security/ProfileActivitySecurityTest.java:27]: So here, you test (as in the other tests) that the button button_login is not hidden, correct?

File: app/src/main/java/com/sdp/movemeet/EditProfileActivity.java [hunk 7]
```diff
@@ -67,6 +67,9 @@ protected void onCreate(Bundle savedInstanceState) {
             userId = user.getUid();
             storageReference = FirebaseStorage.getInstance().getReference();
             loadRegisteredUserProfilePicture(userId);
+        } else {
+            startActivity(new Intent(getApplicationContext(), LoginActivity.class)); // sending the user to the "Login" activity
+            finish();
```
Comment (Antho1426) [inline app/src/main/java/com/sdp/movemeet/EditProfileActivity.java:72]: Yes, good redirection, this is important everywhere in fact, right!

Comment (jlstoupy98) [reply app/src/main/java/com/sdp/movemeet/EditProfileActivity.java:72]: Yes, that the main point of my modifications for this week. On each page requires a connection to be accessed, I added this test. This way, if a user ends up on one of the pages requiring a connection when he is not connected, he will be redirected to the login page

File: app/src/androidTest/java/com/sdp/movemeet/Security/ProfileActivitySecurityTest.java (hunk 6 above)
Comment (jlstoupy98) [reply app/src/androidTest/java/com/sdp/movemeet/Security/ProfileActivitySecurityTest.java:27]: Sort of. Actually, this test is here to test the redirection of unlogged user to login page. So in all these tests, I try to instantiate a class (for example in this case the profile page) without logging in. If the redirection works, we are supposed to be on the Login page and to test that, I look if the "login" button is displayed. If this is the case, it means that the security has worked.

File: app/src/androidTest/java/com/sdp/movemeet/Navigation/ActivityDescriptionNavigationTest.java (hunk 4 above)
Comment (jlstoupy98) [reply app/src/androidTest/java/com/sdp/movemeet/Navigation/ActivityDescriptionNavigationTest.java:112]: Yes but it's more complete now. I created test set for each page and for each page, in this tests, I just test all the redirection links of the menu. So, it's a complete test for the navigation part (and a little bit long)

File: app/src/androidTest/java/com/sdp/movemeet/Navigation/ChatActivityNavigationTest.java (hunk 5 above)
Comment (jlstoupy98) [reply app/src/androidTest/java/com/sdp/movemeet/Navigation/ChatActivityNavigationTest.java:3]: Exactly !

File: app/src/androidTest/java/com/sdp/movemeet/Login/LoginActivityTest.java (hunk 3 above)
Comment (jlstoupy98) [reply app/src/androidTest/java/com/sdp/movemeet/Login/LoginActivityTest.java:91]: I think it's just to remove the blank spaces 😂 and change the Login_TrueNonEmpty test that was commented out because it wasn't working. But now everything is fine!

File: app/src/androidTest/java/com/sdp/movemeet/Activity/UploadActivityActivityTest.java (hunk 2 above)
Comment (jlstoupy98) [reply app/src/androidTest/java/com/sdp/movemeet/Activity/UploadActivityActivityTest.java:113]: Here, the "testRule.getScenario() " retrieved the scenario created in the @Rule. Except that with my security modifications, instantiating directly a page requiring a connection redirects to the login so no test can be done since we are not on the activity expected by the tests. With this way of doing, there is no more @Rule.We start by connecting in the @Before and only when we are connected, we instantiate the activity we want thanks to "ActivityScenario.launch(UploadActivityActivity.class)"

File: app/src/androidTest/java/com/sdp/movemeet/Activity/ActivityNotNullDescriptionTest.java (hunk 1 above)
Comment (jlstoupy98) [reply app/src/androidTest/java/com/sdp/movemeet/Activity/ActivityNotNullDescriptionTest.java:95]: Yes, it seems to be possible ! Just be aware that I assume that to delete the account you have to be logged in. So the user created in the Register will have to log in before launching the second activity.

Comment (Antho1426) [reply app/src/androidTest/java/com/sdp/movemeet/Activity/ActivityNotNullDescriptionTest.java:95]: Yes right, we will check if my test for deleting the user account passes!

File: app/src/androidTest/java/com/sdp/movemeet/Activity/UploadActivityActivityTest.java (hunk 2 above)
Comment (Antho1426) [reply app/src/androidTest/java/com/sdp/movemeet/Activity/UploadActivityActivityTest.java:113]: Ok I see, thank your for your explanation!

**Review by Antho1426 — APPROVED** [review]

File: app/src/androidTest/java/com/sdp/movemeet/Navigation/MainActivityNavigationTest.java [hunk 8]
```diff
@@ -121,35 +104,16 @@ public void mainActivityToStartActivity() {
...
     public void mainActivityGotoHome() {
-        onView(withId(R.id.nav_view)).perform(NavigationViewActions.navigateTo(R.id.nav_start_activity));
-
-        /*Context context = ApplicationProvider.getApplicationContext();
-
-        Intent activity = new Intent(context, ActivityDescriptionActivity.class);
-        activity.putExtra(ActivityDescriptionActivity.EXTRA_ACTIVITY_ID ,DUMMY_ACTIVITY_ID);
-        activity.putExtra(String.valueOf(ActivityDescriptionActivity.EXTRA_LATITUDE),DUMMY_LATITUDE);
-        activity.putExtra(ActivityDescriptionActivity.EXTRA_DATE ,DUMMY_DATE);
-        activity.putExtra(ActivityDescriptionActivity.EXTRA_TITLE ,DUMMY_TITLE);
```
Comment (kotleta2007) [inline app/src/androidTest/java/com/sdp/movemeet/Navigation/MainActivityNavigationTest.java:132]: Why did you remove this test?

File: app/src/androidTest/java/com/sdp/movemeet/Activity/ActivityNotNullDescriptionTest.java [hunk 9]
```diff
@@ -54,13 +69,51 @@
...
+    public void signIn() {
+        CountDownLatch latch = new CountDownLatch(1);
+
+        mAuth = FirebaseAuth.getInstance();
+
+        mAuth.signInWithEmailAndPassword("movemeet@gmail.com", "password").addOnCompleteListener(new OnCompleteListener<AuthResult>() {
+            @Override
+            public void onComplete(@NonNull Task<AuthResult> task) {
+                if (task.isSuccessful()) {
+                    sleep(500);
```
Comment (RoxaneBurri) [inline app/src/androidTest/java/com/sdp/movemeet/Activity/ActivityNotNullDescriptionTest.java]: why do you need sleep where ?

File: app/src/androidTest/java/com/sdp/movemeet/Profile/ProfileActivityTest.java [hunk 10]
```diff
@@ -73,7 +73,10 @@ public void editProfileActivity_getsCorrectData() {
             onView(withId(R.id.edit_text_edit_profile_phone)).check(matches(withText(TEST_PHONE)));
             onView(withId(R.id.edit_text_edit_profile_description)).check(matches(withText(TEST_DESCRIPTION)));
         }
-    }
+    }*/
+
+    @Test
+    public void truc(){}
```
Comment (kotleta2007) [inline app/src/androidTest/java/com/sdp/movemeet/Profile/ProfileActivityTest.java:59]: What is the purpose of `truc()` ?

File: app/src/androidTest/java/com/sdp/movemeet/Navigation/MainActivityNavigationTest.java [hunk 11]
```diff
@@ -58,28 +61,14 @@
 
 @RunWith(AndroidJUnit4.class)
 
-public class MainActivityTest {
```
Comment (RoxaneBurri) [inline app/src/androidTest/java/com/sdp/movemeet/Navigation/MainActivityNavigationTest.java:61]: yes you can delete, it was a test

**Review by RoxaneBurri — APPROVED** [review]

**Review by kotleta2007 — APPROVED** [review]

File: app/src/androidTest/java/com/sdp/movemeet/Activity/ActivityNotNullDescriptionTest.java (hunk 9 above)
Comment (jlstoupy98) [reply app/src/androidTest/java/com/sdp/movemeet/Activity/ActivityNotNullDescriptionTest.java]: I don't know if it's really necessary, maybe I can delete it !

File: app/src/androidTest/java/com/sdp/movemeet/Navigation/MainActivityNavigationTest.java (hunk 8 above)
Comment (jlstoupy98) [reply app/src/androidTest/java/com/sdp/movemeet/Navigation/MainActivityNavigationTest.java:132]: According to roxane's message just above, it is not necessary so I will delete it

File: app/src/androidTest/java/com/sdp/movemeet/Profile/ProfileActivityTest.java (hunk 10 above)
Comment (jlstoupy98) [reply app/src/androidTest/java/com/sdp/movemeet/Profile/ProfileActivityTest.java:59]: Haha is was just an empty test because all other test in the class were commented and we need a test in a class to avoid error ;)

File: app/src/androidTest/java/com/sdp/movemeet/Navigation/MainActivityNavigationTest.java (hunk 8 above)
Comment (kotleta2007) [reply app/src/androidTest/java/com/sdp/movemeet/Navigation/MainActivityNavigationTest.java:132]: Ok
